### PR TITLE
If the user do theano.function(profile=True), do a warning about CUDA…

### PR DESCRIPTION
**Pull request by nouiz**
…_LAUNCH_BLOCKING not being used.

fix gh-5669

### CODE REVIEW

File: theano/gof/fg.py [hunk 1]
```diff
@@ -16,7 +15,6 @@
 from theano import config
 
 from six import iteritems, itervalues
-from six.moves import StringIO
```
Comment (lamblin) [inline theano/gof/fg.py]: These changes actually make flake8 fail. This should probably go in a different PR.

File: theano/compile/profiling.py [hunk 2]
```diff
@@ -253,20 +254,25 @@ def reset(self):
...
+                " Theano GPU ops execution is asynchronous by default."
+                " So by default, the profile is useless."
+                " You must set the environment variable"
+                " CUDA_LAUNCH_BLOCKING to 1 to tell the CUDA driver to"
+                " synchronize the execution to get a meaningful profile.")
+            if config.profile:
+                raise Exception(msg)
+            else:
+                warnings.warn(msg)
+
```
Comment (lamblin) [inline theano/compile/profiling.py]: OK with that part.

**Comment by nouiz** [comment]
I remobed the commit that was in the wrong PR. travis passed, only jenkins is missing.
